agent/src/beerocks/slave/task_pool.h

```diff
@@ -0,0 +1,39 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TASK_POOL_H_
+#define _TASK_POOL_H_
+
+#include "task.h"
+
+#include <beerocks/tlvf/beerocks_message_action.h>
+
+namespace son {
+
+class task_pool {
+
+public:
+    task_pool() {}
+    ~task_pool() {}
+
+    bool add_task(std::shared_ptr<task> new_task);
+    bool is_task_running(int id);
+    void kill_task(int id);
+    void push_event(int task_id, int event_type, void *obj = nullptr);
+    void response_received(std::string mac,
+                           std::shared_ptr<beerocks::beerocks_header> beerocks_header);
```

Comment:
Actually I have decided to remove the task infrastructure from the DB before you commented it, because I realized it does not work well with the needs of the UA.
So sorry you put so much effort into this comment, I removed the task from the DB.